client/shared/src/test/scala/org/http4s/client/ClientSuite.scala

```diff
@@ -194,24 +195,26 @@ class ClientSpec extends Http4sSuite with Http4sDsl[IO] {
 
   }
 
-  test("mock client should fail if the internal channel is not synchronous") {
+  test("mock client should not read the body eagerly") {
 
     def app(compiled: Ref[IO, Int]) = HttpApp[IO] { (_: Request[IO]) =>
       Response[IO]()
         .withEntity("foo")
-        .pipeBodyThrough(_.onFinalize(compiled.update(_ + 1)))
+        .pipeBodyThrough(_ ++ fs2.Stream.exec(compiled.update(_ + 1)))
```

Comment:
I was thinking the `exec` should come first, did the test not pass like this?

Reply:
Yep, it didn't pass

Reply:
Interesting! So I guess the `exec` is executed eagerly then, did not know that.